### PR TITLE
Profile Settings Cleanup [Mobile-app changes]

### PR DESCRIPTION
**Pull request by jatzz10**
**Initiative:** 

> Profile Settings Cleanup

**Changes done:**

> - Modified the content of few of the nav-bar items and profile settings page
> - Changed the order of few items in the nav-bar and profile settings page
> - Changed the icons of nav-bar items: "Settings" (initially named as "Profile") and "My Organizations" (initially named as "Switch Accounts")

**Screenshots:**

Screenshot 2021-02-22 at 9 22 01 PM
Screenshot 2021-02-22 at 9 04 54 PM
Screenshot 2021-02-22 at 9 06 09 PM

### CODE REVIEW

File: src/app/auth/switch-org/switch-org.page.html [hunk 1]
```diff
@@ -12,7 +12,7 @@
 <ion-content>
   <div>
     <div class="switch-org--header">
-      Choose Organization
+      Select an account to switch to it
```
Comment (jatzz10) [inline src/app/auth/switch-org/switch-org.page.html:15]: Sai's comment -> Change this to 'Select an organization to switch to it'

Comment (jatzz10) [reply src/app/auth/switch-org/switch-org.page.html]: this is done

File: src/app/fyle/my-profile/my-profile.page.html [hunk 2]
```diff
@@ -201,29 +134,84 @@
...
-        <div class="my-profile--cards-content  my-profile--toggle-container">
-          <div class="my-profile--toggle-label my-profile--toggle-label__full">
-            Please login to the web application at <a (click)="openWebAppLink('app')">{{ROUTER_API_ENDPOINT}}</a> to change mileage settings.
+  <ng-container *ngIf="myETxnc$|async as myETxnc">
+    <div class="my-profile--blocks">
+      <span class="my-profile--blocks-header">Summary</span>
+      <div class="my-profile--card my-profile--card__action">
+        <ng-container *ngIf="eou$|async as eou">
+          <div class="my-profile--card-title">
+            <div class="my-profile--card-title__using-fyle">
```
Comment (khnamrata) [inline src/app/fyle/my-profile/my-profile.page.html:172]: is this ok? "card-title__using-fyle"

Comment (jatzz10) [reply src/app/fyle/my-profile/my-profile.page.html:172]: This is an existing div. I have just rearranged the order of few divs in this html file. And yes, this is ok. In **my-profile.page.scss** file, we have few of these classes following the same naming rule.

```
&--card-title {
    font-size: 16px;
    font-weight: 500;
    color: #4a4a4a;

    &__using-fyle {
      margin-bottom: 16px;
    }

    &__lite {
      color: #ababab;
    }
  }
```

File: src/app/fyle/delegated-accounts/delegated-accounts.page.html [hunk 3]
```diff
@@ -12,7 +12,7 @@
 <ion-content>
   <div>
     <div class="delegated--header-bar">
-      Choose Delegated Account
+      Select an account to switch to it
```
Comment (khnamrata) [inline src/app/fyle/delegated-accounts/delegated-accounts.page.html:15]: shouldn't it be "Select an account to switch to"

**Review by khnamrata — APPROVED** [review]

File: src/app/fyle/delegated-accounts/delegated-accounts.page.html (hunk 3 above)
Comment (jatzz10) [reply src/app/fyle/delegated-accounts/delegated-accounts.page.html:15]: Yeah, this content was given by Sai only. Will check again with him regarding this.

File: src/app/fyle/ccc-classify-actions/ccc-classify-actions.page.html [hunk 4]
```diff
@@ -47,7 +47,7 @@
 
     <div class="ccc-classify-actions--action" (click)="openMarkPersonalPopover(cccExpense)" *ngIf="cccExpense.amount >= 0" matRipple>
       <mat-icon  class="ccc-classify-actions--icon" svgIcon="fy-individual"></mat-icon>
-      <div >Classing as Personal</div>
+      <div >Classify as Personal</div>
     </div>
```
Comment (jatzz10) [inline src/app/fyle/ccc-classify-actions/ccc-classify-actions.page.html:51]: Found this typo. Have fixed this bug:
Screenshot 2021-02-23 at 10 58 40 PM

File: src/app/fyle/ccc-classify-actions/ccc-classify-actions.page.ts [hunk 5]
```diff
@@ -91,7 +91,7 @@ export class CccClassifyActionsPage implements OnInit {
         '        will collect this amount from you based on your company\'s policy. Do you wish to classify this as a\n' +
         '        personal transaction?',
       primaryCta: {
-        text: 'Classing as Personal'
+        text: 'Classify as Personal'
```
Comment (jatzz10) [inline src/app/fyle/ccc-classify-actions/ccc-classify-actions.page.ts:94]: Same comment as above

File: src/app/app.component.html [hunk 6]
```diff
@@ -45,7 +45,7 @@
                 </a>
               </ion-item>
               <ng-container *ngIf="allowedActions$|async as allowedActions">
-                <div *ngIf="sideMenuItem.title === 'Profile' && (allowedActions?.allowedReportsActions?.approve || allowedActions?.allowedAdvancesActions?.approve)" class="sidemenu--divider"></div>
+                <div *ngIf="sideMenuItem.title === 'Advances' && (allowedActions?.allowedReportsActions?.approve || allowedActions?.allowedAdvancesActions?.approve)" class="sidemenu--divider"></div>
```
Comment (madhavmansuriya40) [inline src/app/app.component.html:48]: what if the advance is disabled from org will this condition still work?

File: src/app/app.component.ts [hunk 7]
```diff
@@ -439,19 +419,39 @@ export class AppComponent implements OnInit {
...
           },
+          {
+            title: 'Delegated Accounts',
+            isVisible: isDelegatee && !this.isSwitchedToDelegator,
+            icon: 'fy-delegate-switch',
+            route: ['/', 'enterprise', 'delegated_accounts'],
+            disabled: true
+          },
+          {
+            title: 'Switch back to my account',
```
Comment (madhavmansuriya40) [inline src/app/app.component.ts:442]: please test if the text overflow is working properly

File: src/app/fyle/delegated-accounts/delegated-accounts.page.html (hunk 3 above)
Comment (madhavmansuriya40) [reply src/app/fyle/delegated-accounts/delegated-accounts.page.html:15]: please confirm and update here, if this is ok

File: src/app/app.component.ts [hunk 8]
```diff
@@ -439,19 +419,39 @@ export class AppComponent implements OnInit {
             disabled: true
           },
           {
-            title: 'Help',
+            title: 'Settings',
```
Comment (madhavmansuriya40) [inline src/app/app.component.ts:422]: isn't it difficult to imagine log out under the settings tab? 😅 

if it is intentionally done, its ok

Comment (jatzz10) [reply src/app/app.component.ts:422]: Yeah, this is intentional.

**Review by madhavmansuriya40 — APPROVED** [review]
please resolve the comments, the rest looks good to me

File: src/app/fyle/delegated-accounts/delegated-accounts.page.html (hunk 3 above)
Comment (jatzz10) [reply src/app/fyle/delegated-accounts/delegated-accounts.page.html]: This has been changed to `Select the account you wish to manage` - discussed with Sai.

File: src/app/app.component.html (hunk 6 above)
Comment (jatzz10) [reply src/app/app.component.html]: This has been resolved thanks to @madhavmansuriya40 :)

File: src/app/app.component.ts (hunk 7 above)
Comment (jatzz10) [reply src/app/app.component.ts:443]: Already checked. Not overflowing.

File: src/app/app.component.ts [hunk 9]
```diff
@@ -266,12 +267,6 @@ export class AppComponent implements OnInit {
             icon: 'fy-reports-new',
             route: ['/', 'enterprise', 'my_reports']
           },
-          {
-            title: 'Advances',
-            isVisible: orgSettings.advances.enabled || orgSettings.advance_requests.enabled,
-            icon: 'fy-advances-new',
-            route: ['/', 'enterprise', 'my_advances']
-          },
```
Comment (jatzz10) [inline src/app/app.component.ts:274]: Haven't removed anything here. Just have changed the order. Advances has been placed under Trips in the sidenav now.

File: src/app/app.component.ts [hunk 10]
```diff
@@ -280,22 +275,10 @@ export class AppComponent implements OnInit {
...
-            title: 'Switch to own account',
-            isVisible: this.isSwitchedToDelegator,
-            icon: 'fy-switch',
-            route: ['/', 'enterprise', 'delegated_accounts', {switchToOwn: true}]
-          },
-          {
-            title: 'Profile',
-            isVisible: true,
-            icon: 'fy-profile-new',
-            route: ['/', 'enterprise', 'my_profile']
```
Comment (jatzz10) [inline src/app/app.component.ts:298]: This is the new order of sidenav items:
- Dashboard
- Expenses
- Cards
- Reports
- Trips
- Advances
-------------
- Team Reports
- Team Trips
- Team Advances

- Live Chat
- Settings
- My Organizations
- Delegated Accounts
- Help

File: src/app/app.component.html [hunk 11]
```diff
@@ -45,7 +45,7 @@
                 </a>
               </ion-item>
               <ng-container *ngIf="allowedActions$|async as allowedActions">
-                <div *ngIf="sideMenuItem.title === 'Profile' && (allowedActions?.allowedReportsActions?.approve || allowedActions?.allowedAdvancesActions?.approve)" class="sidemenu--divider"></div>
+                <div *ngIf="sideMenuItem.title === dividerTitle && (allowedActions?.allowedReportsActions?.approve || allowedActions?.allowedAdvancesActions?.approve)" class="sidemenu--divider"></div>
```
Comment (jatzz10) [inline src/app/app.component.html:48]: To understand this change, please refer to the code from file **src/app/app.component.ts** from line no. **459** to **470**.

File: src/app/app.component.ts [hunk 12]
```diff
@@ -328,17 +311,35 @@ export class AppComponent implements OnInit {
             openHelp: true
           },
           {
-            title: 'Help',
+            title: 'Settings',
             isVisible: true,
-            icon: 'fy-help-new',
-            route: ['/', 'enterprise', 'help']
+            icon: 'fy-settings-o',
```
Comment (iamtarun1993) [inline src/app/app.component.ts:316]: why file name is "fy-settings-o" not "fy-settings" ??

File: src/app/shared/components/delegated-acc-message/delegated-acc-message.component.html [hunk 13]
```diff
@@ -1,3 +1,3 @@
 <span class="delegated-acc">
-  Inside {{ delegateeName | ellipsis: 20 }}'s Delegated Account
+  You're now managing {{ delegateeName | ellipsis: 20 }}'s account
```
Comment (iamtarun1993) [inline src/app/shared/components/delegated-acc-message/delegated-acc-message.component.html:2]: Here content size is increased, please check ellipsis 20 is ok or should we need to reduce it.
Please check this on iPhone 5 in browser

File: src/app/fyle/my-profile/my-profile.page.html [hunk 14]
```diff
@@ -201,29 +134,84 @@
...
     </div>
   </div>
 
-  <ng-container *ngIf="eou$|async as eou">
-    <ng-container *ngIf="org$|async as org">
-      <div class="my-profile--card my-profile--card__action" *ngIf="!org.lite && eou.ou.roles.indexOf('ADMIN') > -1">
-        <div class="my-profile--card-title">Mileage Logs</div>
-        <div class="my-profile--cards-content  my-profile--toggle-container">
-          <div class="my-profile--toggle-label my-profile--toggle-label__full">
-            Please login to the web application at <a (click)="openWebAppLink('app')">{{ROUTER_API_ENDPOINT}}</a> to change mileage settings.
```
Comment (iamtarun1993) [inline src/app/fyle/my-profile/my-profile.page.html:222]: Are we removing this block fully from the mobile app??

File: src/app/shared/components/delegated-acc-message/delegated-acc-message.component.html (hunk 13 above)
Comment (jatzz10) [reply src/app/shared/components/delegated-acc-message/delegated-acc-message.component.html:2]: Right, I have reduced the `ellipsis` from **20** to **15** and tested for iPhone5 in browser.
Here is the comparison:
- With `ellipsis: 20`:
   
   Screenshot 2021-02-25 at 1 11 41 PM

- With `ellipsis: 15`:

   Screenshot 2021-02-25 at 1 08 33 PM

File: src/app/fyle/my-profile/my-profile.page.html (hunk 14 above)
Comment (jatzz10) [reply src/app/fyle/my-profile/my-profile.page.html:222]: Yes sir, this is intentional.

File: src/app/app.component.ts [hunk 15]
```diff
@@ -439,21 +420,54 @@ export class AppComponent implements OnInit {
...
+        - else if 'Trips' is enabled, the divider will be placed under 'Trips',
+        - else it will be placed under 'Reports'
+      */
+      this.dividerTitle = 'Reports';
+      if (orgSettings.trip_requests.enabled && (!orgSettings.trip_requests.enable_for_certain_employee || (orgSettings.trip_requests.enable_for_certain_employee && orgUserSettings.trip_request_org_user_settings.enabled))) {
+        this.dividerTitle = 'Trips';
+      }
+      if (orgSettings.advances.enabled || orgSettings.advance_requests.enabled) {
+        this.dividerTitle = 'Advances';
+      }
```
Comment (jatzz10) [inline src/app/app.component.ts:470]: Screenshot 2021-02-25 at 3 09 15 PM

File: src/app/auth/switch-org/switch-org.page.html [hunk 16]
```diff
@@ -12,7 +12,7 @@
 <ion-content>
   <div>
     <div class="switch-org--header">
-      Choose Organization
+      Select an Organization to switch to it
```
Comment (abhishek1234321) [inline src/app/auth/switch-org/switch-org.page.html:15]: `Select an Organization to switch to` might be better no? `to it` sounds a little odd

File: src/app/fyle/my-profile/my-profile.page.html [hunk 17]
```diff
@@ -48,7 +48,7 @@
 
         <div *ngIf="eou.ou.employee_id" class="my-profile--mobile my-profile--card-item">
           <div class="my-profile--mobile-label">
-            ID
+            Emp. ID
```
Comment (abhishek1234321) [inline src/app/fyle/my-profile/my-profile.page.html:51]: what is this?

File: src/app/shared/components/delegated-acc-message/delegated-acc-message.component.html [hunk 18]
```diff
@@ -1,3 +1,3 @@
 <span class="delegated-acc">
-  Inside {{ delegateeName | ellipsis: 20 }}'s Delegated Account
+  You're now managing {{ delegateeName | ellipsis: 15 }}'s account
```
Comment (abhishek1234321) [inline src/app/shared/components/delegated-acc-message/delegated-acc-message.component.html:2]: why change it to 15?

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

File: src/app/fyle/my-profile/my-profile.page.html (hunk 17 above)
Comment (jatzz10) [reply src/app/fyle/my-profile/my-profile.page.html:51]: Employee Id

File: src/app/app.component.ts [hunk 19]
```diff
@@ -328,17 +311,35 @@ export class AppComponent implements OnInit {
...
+            title: 'Settings',
             isVisible: true,
-            icon: 'fy-help-new',
-            route: ['/', 'enterprise', 'help']
+            icon: 'fy-settings',
+            route: ['/', 'enterprise', 'my_profile']
           },
           {
-            title: 'Switch Accounts',
+            title: 'My Organizations',
```
[the remark this answers is not in the document]
Comment (jatzz10) [reply src/app/app.component.ts]: This has been discussed

File: src/app/auth/switch-org/switch-org.page.html (hunk 16 above)
Comment (jatzz10) [reply src/app/auth/switch-org/switch-org.page.html]: This has been discussed

File: src/app/shared/components/delegated-acc-message/delegated-acc-message.component.html (hunk 18 above)
Comment (jatzz10) [reply src/app/shared/components/delegated-acc-message/delegated-acc-message.component.html:2]: Tarun suggested to reduce it a bit in his comment [here](https://github.com/fylein/fyle-mobile-app2/pull/266#discussion_r582554325)

File: src/app/app.component.ts [hunk 20]
```diff
@@ -385,25 +379,12 @@ export class AppComponent implements OnInit {
...
-            icon: 'fy-delegate-switch',
-            route: ['/', 'enterprise', 'delegated_accounts'],
-            disabled: true
-          },
-          {
-            title: 'Switch to own account',
-            isVisible: this.isSwitchedToDelegator,
-            icon: 'fy-switch',
-            route: ['/', 'enterprise', 'delegated_accounts', {switchToOwn: true}],
+            title: 'Advances',
```
[the remark this answers is not in the document]
Comment (jatzz10) [reply src/app/app.component.ts:382]: Yes, sir. It is intentional. This is how this if-else block has been coded before.
```
 if (isConnected) {
   this.sideMenuList = [
     { ..... },
     { ..... },
     ....
 } else {
   this.sideMenuList = [
     { ..... },
     { ..... },
     ....
 }
```

File: src/app/app.component.ts [hunk 21]
```diff
@@ -439,21 +420,54 @@ export class AppComponent implements OnInit {
...
             isVisible: true,
-            icon: 'fy-help-new',
-            route: ['/', 'enterprise', 'help'],
-            disabled: true
+            icon: 'fy-settings',
+            route: ['/', 'enterprise', 'my_profile'],
           },
           {
-            title: 'Switch Accounts',
+            title: 'My Organizations',
```
[the remark this answers is not in the document]
Comment (jatzz10) [reply src/app/app.component.ts]: Same comment as [this](https://github.com/fylein/fyle-mobile-app2/pull/266/files#r585585128)

File: src/app/fyle/my-profile/my-profile.page.html (hunk 17 above)
Comment (abhishek1234321) [reply src/app/fyle/my-profile/my-profile.page.html:51]: if we have enough space please change it to Employee ID, if not ignore

**Review by abhishek1234321 — APPROVED** [review]

**Pull request merged**